### PR TITLE
[CR] Learning to read

### PR DESCRIPTION
**Pull request by Kelenius**
#### Summary
```SUMMARY: Content "Split illiteracy into two traits, one of them allows to learn to read."```

#### Purpose of change
Illiterate trait always seemed odd to me - sure, maybe you can't read, but what's stopping you from learning to? What does it represent?

#### Describe the solution
Illiterate trait was split into two: one of them works exactly like the old one and covers your principal inability to learn how to read. The second one only gives 2 points (vs 6), but if you find a grammar book (a new item, can be found in most places children book can be found) and read it, it will be replaced with Slow Reader (1 point).

The new trait is called "Reading Inability", which isn't a name I'm happy with, but I can't come up with anything better - I've considered "Dyslexia", but dyslexic people can learn to read. "Agraphia" was something I also considered, but that seems to be more specific than I want.

The codes are also now mixed up, ILLITERATE is "Reading Inability", and CANNOT_READ is "Illiterate", because otherwise old characters would have their traits changed.

### CODE REVIEW

**Comment by Brambor** [comment]
Really cool. I cannot quite read book times, does it take a REALLY long time to learn to read? It should takes one year [no sources, I might be wrong] for a kid to learn to read. Additionally it might turn in beggining reader that reads at 1/10 the speed of normal reader, as childerns that just learned [no sources, I might be wrong]. And then the same book read again and again, or some other book or any book could improve beginer reader into either slow or normal(!) reading capacity. Representing, that, as child, the person learned to read and might be good at it as others are.

**Comment by MT-Arnoldussen** [comment]
Instantly learning to read from a  single instruction book does not seem any more realistic or sensible than never being able to read at all.

**Comment by tinukedaya** [comment]
It's about as realistic as learning kungfu by reading a single booklet. 
But as said already. Still better than not being able to learn it at all.

What I would suggest though is make change to slow reader chance related and tied to character inteligence. 

You can be really smart but maybe just didn't have the chance to learn to read. In that case learning it from the book should be quite easy.

And of course if the character i just really stupid it should

**Comment by lucasmr** [comment]
Maybe require a NPC+book to teach it? Don't know how it would be implemented, but reading and martial arts could be made to require a teacher.

File: src/player.h [hunk 1]
```diff
@@ -355,6 +355,8 @@ class player : public Character
         bool has_alarm_clock() const;
         /** Returns true if the player or their vehicle has a watch */
         bool has_watch() const;
+		/** Returns true if the player is illiterate */
```
Comment (Night-Pryanik) [inline src/player.h]: Tabs.

**Comment by Night-Pryanik** [comment]
It could probably be done as slow training of a new hidden skill. After one reaches first level of this skill, he is considered literate.

File: src/player.h (hunk 1 above)
Comment (Kelenius) [reply src/player.h]: Stupid VS, I thought I fixed all of these, but nope.

**Comment by kevingranade** [comment]
I'm against it, if you've operated your whole life pre-cataclysm without learning to read, it seems bizarre that now that it's much more difficult and has a lower payoff, youd suddenly decide to put in the work and learn.

From a game point of view, taking a very severe negative trait and then grinding it away very much sounds like powergaming.

I also agree with the feedback that the implementation trivializes the process, a process that normally takes years of study is reduced to a single interaction with a book, even if I weren't against this process in general, this would be an insufficient implementation.

**Comment by Brambor** [comment]
> I'm against it, if you've operated your whole life pre-cataclysm without learning to read, it seems bizarre that now that it's much more difficult and has a lower payoff, youd suddenly decide to put in the work and learn.

PC may think _There is a lot of information I need to get to. And I know this knowledge is in those books. If I want better life I need to learn to read._ or _Now that everyone is dead, humans exist only in text, I need to speak to them._ or _Now I finally can educate myself, as I was farmboy that didn't go beyond farm plots._

> From a game point of view, taking a very severe negative trait and then grinding it away very much sounds like powergaming.

Currently I play with random character (If I really dislike them, I reroll them) in a new world, when they die, I delete the world and start anew. I would not mind this opportunity/challenge of not being able to read but being able to learn it. I reroll any ILLITERATE character right away, because I like books in this game to much to give up on them in playthrougth that may just be really long. (I should try one too, though.)

**Comment by Kelenius** [comment]
>It could probably be done as slow training of a new hidden skill. After one reaches first level of this skill, he is considered literate.

This is an interesting idea, and hidden skills could probably be used in other places later.

>I'm against it, if you've operated your whole life pre-cataclysm without learning to read, it seems bizarre that now that it's much more difficult and has a lower payoff, youd suddenly decide to put in the work and learn.

Reading is very useful after the cataclysm - if it wasn't, this trait wouldn't have been such a big deal. As for the reasons why you never learned to before... google gives mixed results, but it seems that about 15% of adults in modern day USA can't read English.

>From a game point of view, taking a very severe negative trait and then grinding it away very much sounds like powergaming.

I see your point, but you're only replacing a -2 trait with a -1 trait, so you only gain one point compared to just taking slow reader, while making your early game harder. There's a precedent for this in the game already - you can take both near-sightedness and far-sightedness and then install a bionic to fix them - that takes more time to fix, but also gives more points.

>I also agree with the feedback that the implementation trivializes the process, a process that normally takes years of study is reduced to a single interaction with a book, even if I weren't against this process in general, this would be an insufficient implementation.

This is true. I'll think of a way to make this process more immersive. I don't think it should take years - education in CDDA is generally compressed, you can't go from knowing nothing to master of electronics after reading some books for three days - but learning to read should probably take several stages.

NPCs teaching the player is an interesting idea but I'm just not sure if I can tackle it... I'll look into it.

**Comment by kevingranade** [comment]
> I don't think it should take years

It should take at least a year, that's already "compressed".  If you're not willing to make that happen, it's not getting merged.

**Comment by Kelenius** [comment]
If we're talking about learning it that slow, it could make sense to create a new skill for reading, that would also cover reading speed in general, and could work similarly to the intelligence requirement on the books, so on the low skill level you could read simple books okay-ish, but studying quantum physics will be slow as molasses. Slow reader and fast reader could be rolled into that skill as well. But why a year? That's much longer than it takes to learn other skills.

**Comment by tinukedaya** [comment]
I really think you are exagerating the difficulty of learning how to read. All the illiterate people in the world are not illiterate because it's difficult, but because they didn't have the opportunity to learn.

One of my kids learned to write and read on her own even before school. All i had to was to tell her how a letter looks like when she came asking. 

So I really think it should just be tied to inteligence and be longer process than just skim through a book once, but it should be possible.

And the motivation after cataclysm would be really high

**Comment by Night-Pryanik** [comment]
> but studying quantum physics will be slow as molasses

The problem of studying quantum physics is the difficulty of understanding this science, not the difficulty of reading the words or formulas in the books.

**Comment by Kelenius** [comment]
>The problem of studying quantum physics is the difficulty of understanding this science, not the difficulty of reading the words or formulas in the books.

It's both. Reading technical English is harder than fiction.

**Comment by Night-Pryanik** [comment]
I just wanted to say that even if I manage to quickly read the book of quantum physics, it doesn't mean that I understood what I've read.

**Comment by chaisawlajatang** [comment]
I'm against it. 
1. Illiterate trait is a _challenge_, you pick it to make a game much harder and gain some extra points at the start. If you don't like it, pick Slow reader instead.
2. Learning to read (at the level of a kid) does NOT mean an ability to understand complex technical texts.
3. Learning to read should take months for an adult and involve somebody's help.

**Comment by Brambor** [comment]
@chaisawlajatang 
> Illiterate trait is a challenge, you pick it to make a game much harder and gain some extra points at the start. If you don't like it, pick Slow reader instead.

It is not deleting Illiterate trait nor allowing it to learn to read but splits it into two, so it adds another version of that trait, that can learn to read. For balance reasons, you get less points for picking that trait.

> Learning to read (at the level of a kid) does NOT mean an ability to understand complex technical texts.

Neither does it in cataclysm. Every character without Illiterate trait can read, but can understand the books only if they have enough skill, which represents knowleadge, in given technical text.
Given that every book would explain its terms in some kind of index, then it makes sense that with given skill you will understand given book.

> Learning to read should take months for an adult and involve somebody's help.

That is argument that has been said and has been spoken about in comments in here. It looks to be in favor of what you are asking for.

All in all, I take your comment as resolved.

**Comment by ShinQuickMan** [comment]
Maybe the new trait could be described as not an inability to read outright, but an inability to read perfect English. Could be that the PC is foreign-born, and for whatever reason, never got around to learning how to read and write in English (though they could speak it well enough). Happens all the time IRL.

**Pull request closed without merging**